internal/v2/service/service.go

```diff
@@ -4,22 +4,26 @@ import (
 	"github.com/babylonlabs-io/staking-api-service/internal/shared/config"
 	dbclients "github.com/babylonlabs-io/staking-api-service/internal/shared/db/clients"
 	"github.com/babylonlabs-io/staking-api-service/internal/shared/http/clients"
-	"golang.org/x/sync/singleflight"
+	"github.com/babylonlabs-io/staking-api-service/internal/shared/services/service"
 )
 
 type V2Service struct {
 	DbClients *dbclients.DbClients
 	Clients   *clients.Clients
 	Cfg       *config.Config
-
-	singleFlightGroup *singleflight.Group
+	service   *service.Service
```

Comment:
This is one thing that I'm not proud of 😄 basically I'm assigning initialized shared service from v1 here. But I'm afraid this is the only way